File: docs-examples/pubsub-java/src/test/java/io/micronaut/configuration/lettuce/docs/pubsub/BookAuditExceptionHandler.java
package io.micronaut.configuration.lettuce.docs.pubsub;

import io.micronaut.configuration.lettuce.pubsub.exception.RedisListenerException;
import io.micronaut.configuration.lettuce.pubsub.exception.RedisListenerExceptionHandler;
import jakarta.inject.Singleton;

@Singleton
class BookAuditExceptionHandler implements RedisListenerExceptionHandler {

    // tag::handler[]
    @Override
    public void handle(RedisListenerException exception) {
        System.err.println("Redis listener failure on " + exception.getMessageChannel() + ": " + exception.getMessage());
    }
    // end::handler[]
}

File: docs-examples/pubsub-kotlin/src/test/kotlin/io/micronaut/configuration/lettuce/docs/pubsub/BookAuditExceptionHandler.kt
package io.micronaut.configuration.lettuce.docs.pubsub

import io.micronaut.configuration.lettuce.pubsub.exception.RedisListenerException
import io.micronaut.configuration.lettuce.pubsub.exception.RedisListenerExceptionHandler
import jakarta.inject.Singleton

@Singleton
class BookAuditExceptionHandler : RedisListenerExceptionHandler {

    // tag::handler[]
    override fun handle(exception: RedisListenerException) {
        System.err.println("Redis listener failure on ${exception.messageChannel}: ${exception.message}")
    }
    // end::handler[]
}
